Source/Core/Core/ConfigLoaders/BaseConfigLoader.cpp

```diff
@@ -43,9 +44,22 @@ void SaveToSYSCONF(Config::LayerType layer)
           const std::string key = info.location.section + "." + info.location.key;
 
           if (setting.type == SysConf::Entry::Type::Long)
+          {
             sysconf.SetData<u32>(key, setting.type, Config::Get(layer, info));
+          }
           else if (setting.type == SysConf::Entry::Type::Byte)
+          {
             sysconf.SetData<u8>(key, setting.type, static_cast<u8>(Config::Get(layer, info)));
+          }
+          else if (setting.type == SysConf::Entry::Type::BigArray)
+          {
+            // Somewhat hacky support for IPL.SADR. The setting only stores the
+            // first 4 bytes even thought the SYSCONF entry is much bigger.
+            SysConf::Entry* entry = sysconf.GetOrAddEntry(key, setting.type);
+            if (entry->bytes.size() < 0x1007 + 1)
+              entry->bytes.resize(0x1007 + 1);
```

Comment:
Not sure this is a good idea, since `BT.CDIF` and `BT.DINF` are BIGARRAY as well; but neither is 0x100A in size.
Have you tried to see what happens if those SYSCONF entries are specified with a size smaller than the previously observed ones?

Reply:
> Not sure this is a good idea, since `BT.CDIF` and `BT.DINF` are BIGARRAY as well; but neither is 0x100A in size.

Yes. I'm not exactly happy with this code, and that's probably the biggest problem with it. I couldn't think of anything better for now, though. It will work entirely correctly since neither `BT.CDIF` nor `BT.DINF` is hooked up to the config system, but it's not good for maintainability.

> Have you tried to see what happens if those SYSCONF entries are specified with a size smaller than the previously observed ones?

No, I haven't tested that.

Reply:
Can't really think of a good way to handle this at this point, aside from hacking in all the known sizes per key and actually implementing how BIGARRAY is supposed to work.
I wonder if we can just get away with only _writing_ those keys if they exist (but not _creating_ them)